packages/react-components/react-table/src/hooks/useKeyboardResizing.ts

```diff
@@ -9,7 +9,7 @@ const PRECISION_MODIFIER = Shift;
 const PRECISION_FACTOR = 1 / 4;
 
 export function useKeyboardResizing(columnResizeState: ColumnResizeState) {
-  const columnId = React.useRef<TableColumnId>();
+  const [columnId, setColumnId] = React.useState<TableColumnId>();
```

Comment:
to avoid a re-render from occuring every time someone selects a different column to resize, you could consider returning a getter function for the column Id which will just get the value from the ref. Or even just return a function to check if the provided column id is the one being resized

When I see the usage of this column id in this PR, I don't see any need for a re-render to occur.